README.md

```diff
@@ -1 +1,291 @@
+
+The core of the Open Canvas specification is extensibility and fallbacks. In the case of one canvas supporting a feature that another canvas doesn't support, we need to enable three things:
+
+1. Canvases need to be able to read and write to an open canvas file without implementing all of the features of all other canvases and also extend the format with features unique to their canvas without breaking other canvases or changing the main specification.
+2. Canvases that lack a feature need to preserve the data for features that they don't implement.
+3. Canvases need to be able to optionally display reasonable fallbacks for features that they don't implement.
+
+Put simply: how do we design the specification to be extensible by individual canvas tools and robust against unknown features?
+
+### Schemas
+
+Schemas are used to define the structure of Nodes and Relations. They provide extensibility by supporting different properties for Nodes and Relations.
+
+Schemas are included inline in the file. In the future, there may be a way to externalize schemas to a remote store.
+
+Schemas have a version number to support schema evolution.
+
+### Extensibility through preserving the property bag
+
+In order to support interchange beteween canvases when features don't overlap, canvases need to preserve nodes and relations that it doesn't support.
+
+In practice, that looks something like this:
+
+1. Canvas A supporting Feature X creates a canvas with a Feature X node in it and exports it as an Open Canvas file.
+2. Canvas B, which does not support Feature X, opens the Open Canvas file, and some edits are made to the canvas.
+3. Canvas B exports the canvas to an Open Canvas file. The nodes for Feature X should still be in the Open Canvas file, unchanged.
+
+NOTE: This is perhaps the most onerous part of supporting Open Canvas; implementing canvases have to store data that they don't need or recognize.
+
+### Extensibility through fallbacks
+
+The base node type supports an optional `fallback` property as a string. If a canvas wishes to visualize a node that it doesn't implement, the canvas can render a fallback element on the canvas at the element's x,y position using the string.
+
+### Extensibility through schema extension
+
+Schemas can extend one another.
+
+When extending, properties are simply merged to establish the full schema. This allows extension with meaningful fallbacks: if an implementer doesn't support the extended schema, but does support the base schema, they can render the node using a simpler representation.
+
+Schemas are defined in JSONSchema format.
+
+```json
+"schemas": [
+  "@ocwg/text": { /* JSONSchema format */ }
+]
+```
+
+For example, a `@tldraw/arrow` schema could extend a `@ocwg/arrow` schema, adding properties that are only applicable to TLDraw arrow.
+
+- [ ] Flesh out this example
```

Comment:
I've added an example of how this might look in https://github.com/orgs/ocwg/discussions/26 .
I think that the schemas should not be an array, but an object.
Maybe the schemas section should be renamed to $defs if it needs to comply with json-schema (see https://json-schema.org/understanding-json-schema/structuring#defs) but I am 100% sure of that...maybe thats only something that should be used in the schema of the spec and not the json data itself.